plugin-server/src/worker/ingestion/lazy-person-container.ts

```diff
@@ -0,0 +1,36 @@
+import { Hub, Person } from '../../types'
+
+export class LazyPersonContainer {
+    teamId: number
+    distinctId: string
+
+    loaded: boolean
+
+    private hub: Hub
+    private promise: Promise<Person | undefined> | null
+
+    constructor(teamId: number, distinctId: string, hub: Hub) {
+        this.teamId = teamId
+        this.distinctId = distinctId
+        this.hub = hub
+        this.promise = null
+        this.loaded = false
+    }
+
+    async get(): Promise<Person | undefined> {
```

Comment:
Instead of being able to return `undefined` and running into other weird issues later, how about we just raise an exception here, so if things don't work we'll have nice stack traces in sentry instead of data being wrong.

Reply:
Data being wrong in what sense? Callsites are expected to handle undefined and they often rely on it for correctness - see e.g. merge logic in person-state.ts

Reply:
Discussed IRL - we'll add a new method to container when the need arises later.